rllib/examples/learners/classes/curiosity_ppo_torch_learner.py

```diff
@@ -0,0 +1,141 @@
+)
+from ray.rllib.connectors.learner.add_next_observations_from_episodes_to_train_batch import (  # noqa
+    AddNextObservationsFromEpisodesToTrainBatch,
+)
+from ray.rllib.core import Columns, DEFAULT_MODULE_ID
+from ray.rllib.core.rl_module.rl_module import RLModuleSpec
+from ray.rllib.examples.rl_modules.classes.inverse_dynamics_model_rlm import (
+    InverseDynamicsModel,
+)
+from ray.rllib.utils.metrics import ALL_MODULES
+
+ICM_MODULE_ID = "_inverse_dynamics_model"
+
+
+class PPOConfigWithCuriosity(PPOConfig):
+    # Define defaults.
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.curiosity_feature_net_hiddens = (256, 256)
+        self.curiosity_feature_net_activation = "relu"
+        self.curiosity_inverse_net_hiddens = (256, 256)
+        self.curiosity_inverse_net_activation = "relu"
+        self.curiosity_forward_net_hiddens = (256, 256)
+        self.curiosity_forward_net_activation = "relu"
+        self.curiosity_beta = 0.2
+        self.curiosity_eta = 1.0
+
+    # Allow users to change curiosity settings.
+    def curiosity(
+        self,
+        *,
+        curiosity_feature_net_hiddens: Tuple[int, ...] = NotProvided,
+        curiosity_feature_net_activation: str = NotProvided,
+        curiosity_inverse_net_hiddens: Tuple[int, ...] = NotProvided,
+        curiosity_inverse_net_activation: str = NotProvided,
+        curiosity_forward_net_hiddens: Tuple[int, ...] = NotProvided,
+        curiosity_forward_net_activation: str = NotProvided,
+        curiosity_beta: float = NotProvided,
+        curiosity_eta: float = NotProvided,
+    ):
+        if curiosity_feature_net_hiddens is not NotProvided:
+            self.curiosity_feature_net_hiddens = curiosity_feature_net_hiddens
+        if curiosity_feature_net_activation is not NotProvided:
+            self.curiosity_feature_net_activation = curiosity_feature_net_activation
+        if curiosity_inverse_net_hiddens is not NotProvided:
+            self.curiosity_inverse_net_hiddens = curiosity_inverse_net_hiddens
+        if curiosity_inverse_net_activation is not NotProvided:
+            self.curiosity_inverse_net_activation = curiosity_inverse_net_activation
+        if curiosity_forward_net_hiddens is not NotProvided:
+            self.curiosity_forward_net_hiddens = curiosity_forward_net_hiddens
+        if curiosity_forward_net_activation is not NotProvided:
+            self.curiosity_forward_net_activation = curiosity_forward_net_activation
+        if curiosity_beta is not NotProvided:
+            self.curiosity_beta = curiosity_beta
+        if curiosity_eta is not NotProvided:
+            self.curiosity_eta = curiosity_eta
+        return self
+
+
+class PPOTorchLearnerWithCuriosity(PPOTorchLearner):
+    def build(self):
+        super().build()
+
+        # Assert, we are only training one policy (RLModule).
+        assert len(self.module) == 1 and DEFAULT_MODULE_ID in self.module
```

Comment:
Would we need to set up a single agent's module to be a MultiRLModule to add curiosity to a single agent? 

In this case it might be more handy to provide Curiosity as a wrapper module that wraps existent RLModules into one that also has curiosity.